resdk/analysis/chip_seq.py

```diff
@@ -17,6 +17,48 @@ def is_background(sample):
     return len(background_relations) > 0
 
 
+def bamsplit(resource, header=None, header2=None):
+    """Run ``Bam split`` process on the resource.
+
+    This method runs `Bam split`_ process on the resource. The process
+    will be run on all samples in the resource. Samples' alignment data
+    object must be valid. Valid alignment data objects are those that
+    were aligned to a hybrid genome with a valid build. Valid builds
+    are: 'hg19_dm6' and 'mm10_dm6'.
+
+    .. _Bam split:
+        http://resolwe-bio.readthedocs.io/en/latest/catalog-definitions.html#process-bam-split
+
+    :param resource: resource on which bam-split will be run
+    :param header: SAM header data object for the primary BAM
+    :type header: `~resdk.resources.data.Data`
+    :param header2: SAM header data object for the secodary BAM
+    :type header: `~resdk.resources.data.Data`
+
+    """
+    inputs = {}
+
+    if header:
+        inputs['header'] = header
+
+    if header2:
+        inputs['header2'] = header2
+
+    results = []
+
+    if not isinstance(resource, list):
+        resource = [resource]
+
+    for single_resource in resource:
+        for sample in get_samples(single_resource):
+            bam = sample.get_bam()
+            inputs['bam'] = bam.id
```

Comment:
Doesn't this work without `.id`? It think you can just write: `inputs['bam'] = sample.get_bam()`.

Reply:
It doesn't work without `.id`. But I can rewrite it as  `inputs['bam'] = sample.get_bam().id`

Reply:
It should work without `.id`. Please check why it doesn't. https://github.com/genialis/resolwe-bio-py/blob/master/resdk/resolwe.py#L292

Reply:
@jvrakor Would you print `type(sample.get_bam()).__name__`? The `get_data_id` function checks if `type(data).__name__ == 'Data'`, so my hypothesis is this condition is not satisfied.